internal/exec/engine.go

```diff
@@ -175,6 +199,26 @@ func (e *Engine) acquireConfig() (cfg types.Config, err error) {
 	return
 }
 
+// acquireConfigOffline only looks at config providers which do not
+// require networking.
+func (e *Engine) detectOfflineConfig() (bool, error) {
+	offlineFetchers := []providers.FuncDetectConfig{
+		cmdline.DetectConfig,
+		file.DetectConfig,
```

Comment:
The `cmdline` provider very much does require networking.  The `file` provider does not, but only runs on the `file` platform, which is only used for testing.

Reply:
Yeah, this is confusing. The goal here isn't to detect "offline" so much as whether a config was provided *at all*.  We know that "statically" with those two by just looking at the filesystem basically.

For other providers we may need to wait for a driver or even do a network request.  *Ideally* we detect the "no config provided" case with those too.  But...that's a higher level thing I think.